src/renderer/views/ProjectsList/subcomponents/ProjectSearch/ProjectSearch.tsx

```diff
@@ -0,0 +1,42 @@
+import { Autocomplete } from '@mui/material';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Publication } from '../../../../../shared/types';
+import SearchField from '../../../../components/SearchField/SearchField';
+
+interface Props {
+  publications: Publication[];
+  onChange: (terms: string[]) => void;
+}
+
+const ProjectSearch: React.FC<Props> = ({ publications, onChange }) => {
+  const { t } = useTranslation();
+  return (
+    <Autocomplete
+      options={getAutocompleteOptions(publications)}
+      onChange={(_, value) => onChange(value)}
+      renderInput={(params) => (
+        <SearchField
+          {...params}
+          InputProps={{
+            'aria-label': t('common.search'),
+            ...params.InputProps,
+          }}
+        />
+      )}
+      size='small'
+      freeSolo
+      multiple
+    />
+  );
+};
+
+export default ProjectSearch;
+
+function getAutocompleteOptions(publications: Publication[]) {
+  const options = [] as string[];
+  publications.forEach(({ tags }) => {
+    tags.forEach((tag) => !options.includes(tag) && options.push(`#${tag}`));
```

Comment:
@LostInStatic this will add all tags as autocomplete options and in result provide items with same keys for React to complain about. This will be due to the fact that `options` array includes items in the format of `#tag_1` and the `tag` itself will be `tag_1` . Therefore, `includes` condition will always throw `false` for duplicate keys.

Reply:
Oops! Amended.